File: 网易2019实习生招聘编程题集合/安置路灯.java
import java.util.*;

public class Main {

	public static Scanner cin = new Scanner(System.in);

	public static void main(String[] args) {
		for (int T = cin.nextInt(); T-- > 0;) {
			int n = cin.nextInt(), ans = 0;
			String str = cin.next();
			for (int i = 0; i < str.length();) {
				for (; i < str.length() && str.charAt(i) == 'X'; ++i) {
				}
				for (; i < str.length() && str.charAt(i) == '.'; ++ans, i += 3) {
				}
			}
			System.out.println(ans);
		}
	}

}

File: 网易2019实习生招聘编程题集合/安置路灯.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
	int T;
	for (cin >> T; T--; ) {
		int n, ans = 0;
		string str;
		cin >> n >> str;
        for (int i = 0; i < (int)str.size(); ) {
			for (; i < (int)str.size() && str[i] == 'X'; ++i) {}
			for (; i < (int)str.size() && str[i] == '.'; ans++, i += 3) {}
        }
        cout << ans << endl;
	}
	return 0;
}
